src/components/Header/index.tsx

```diff
@@ -9,16 +9,16 @@ const Header = (): JSX.Element => {
   const isMobile = useMediaQuery('(max-width: 870px)');
 
   const handleThemeOnClick = (
-    e: MouseEvent<HTMLButtonElement, Event> | KeyboardEvent<HTMLButtonElement>,
     // eslint-disable-next-line @typescript-eslint/ban-types
     toggleTheme: Function,
+    currentTheme: string,
     isKeyPress = false
   ): void => {
     if (isKeyPress) {
       return;
     }
-    const target = e.target as HTMLElement;
-    const toggle = target.innerHTML.includes('nights_stay') ? 'dark' : 'light';
```

Comment:
Issue before was here, the key press was providing the whole `<button />` elm as the target value. I decided to just use the `theme` value provided by the `ThemeToggler` and toggle that value.